packages/graphiql/src/components/QueryHistory.js

```diff
@@ -11,6 +11,9 @@ import PropTypes from 'prop-types';
 import QueryStore from '../utility/QueryStore';
 import HistoryQuery from './HistoryQuery';
 
+const MAX_QUERY_SIZE = 100000;
+const MAX_HISTORY_LENGTH = 20;
```

Comment:
I agree it should be user configurable. I also think that eventually we should have user-definable error handlers so they can decide what to do on quota errors and such, or even make the entire history a separate module so users can decide to save the history and favorites to a database instead of inside the browser, so it persists across browsers or organizations. But that's out of the scope of this PR :smile: